Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ButtonGroup.cpp

```diff
@@ -31,11 +31,22 @@ namespace AzToolsFramework::ViewportUi::Internal
     {
         if (auto buttonEntry = m_buttons.find(buttonId); buttonEntry != m_buttons.end())
         {
-            for (auto& button : m_buttons)
-            {
-                button.second->m_state = Button::State::Deselected;
-            }
+            ClearHighlightedButton();
             buttonEntry->second->m_state = Button::State::Selected;
+            m_highlightedButtonId = buttonId;
+        }
+    }
+
+    void ButtonGroup::ClearHighlightedButton()
+    {
+        if (m_highlightedButtonId == InvalidButtonId)
+        {
+            return;
+        }
+        if (auto buttonEntry = m_buttons.find(m_highlightedButtonId); buttonEntry != m_buttons.end())
```

Comment:
What about having a ref to the highlighted button as a member (instead of the id) to avoid having to do the search each time?

If the button knows its id this is pretty straight forward.

Reply:
the buttons in ```m_buttons``` are unique pointers inside a map, thats why i used the id instead.

Reply:
I see. I guess the member could be an iterator to the map's element. Probably not worth it just to save a search into a map, which is quite fast already.